core/src/components/list-header/test/basic/list-header.e2e.ts

```diff
@@ -12,3 +12,60 @@ configs().forEach(({ title, screenshot, config }) => {
     });
   });
 });
+
+/**
+ * This behavior needs to be tested in both modes and directions to
+ * make sure the safe area padding is applied only to that side
+ * regardless of direction
+ */
+configs().forEach(({ title, screenshot, config }) => {
```

Comment:
Implementation looks good, I just have the same question as in https://github.com/ionic-team/ionic-framework/pull/28372#discussion_r1365789776